### ISSUE TITLE
Validation accuracy not improving

### ISSUE THREAD

**Issue by sfpoveda** [seisbench/seisbench#118]
Hi dears,

I'm checking how well EQTransformer model is doing on my small dataset (~1000 events). I reach a minimum validation loss of 0.11 and a validation accuracy that always stucks in approximately 75% (which is not enough to calculate reliable predictions). Do you have any advice that could help me to improve the validation accuracy? Any advice? Despite validation loss decreases in each epoch, validation accuracy does not. I want to at least reach a validation accuracy 90%+. So, any good comment or advice that could help me to reach this goal is welcome.
Note: I've alerady applied the recommended augmentations to the data.

Below an image of the loss and accuracy I obtained

![image](https://user-images.githubusercontent.com/93283899/180904623-331e1a8a-4a51-4ad4-b6d8-e2b146af3977.png)

**Comment by yetinam** [comment]
Hi @sfpoveda ,
unfortunately, I think 1000 events alone might not get you any better validation loss, just because this is relatively few data. However, I'd recommend you to try out transfer learning, i.e., instead of starting with a randomly initialized model, start with a pretrained one and only adjust it to your data. You can simply do this by initialising your model with `model = sbm.EQTransformer.from_pretrained("stead")`. In principle, you can use any pretrained weights, but there seems to be an issue when starting with the original ones, outline in #98 .

**Comment by sfpoveda** [comment]
To initialise the model with a pretrained weights dataset and then train it on my specific data, what should my code look like? Can you please give me a clue on how to implement this code? Because I need to specify "classes=1", "in_channels=1", phases="P" and a "sampling_rate=500, and  I cannot specify this from the line code "model = sbm.EQTransformer.from_pretrained("stead")". How do I keep the weights of the first layers and change only those of the last layers so that I can use transfer learning?

**Comment by yetinam** [comment]
Hi! Indeed, this will need a bit more of an adjustment. The general idea is to create two models, one with the correct shape and one from the pretrained model. Than you need to build a fitting state dict from the old one. This will require changing the input convolution (one channel instead of three) and removing one of the output branches. The code could look something like this:

```python
target = sbm.EQTransformer(sampling_rate=500, classes=1, in_channels=1, phases="P")
source = sbm.EQTransformer.from_pretraine("instance")

source_state_dict = source.state_dict()
target_state_dict = modify_the_state_dict_accordingly(source_state_dict)  # This is the key step to implement

target.load(target_state_dict)
```

**Comment by sfpoveda** [comment]
Thanks you so much!

**Comment by sfpoveda** [comment]
I expected to have a better accuracy using transfer learning, but surprisingly I got worst accuracy. I mean, without transfer learning I reached a validation accuracy of 80%, but using transfer learning the validation accuracy is 67%. Maybe it is not working as well as I want because my signals are whale calls, which have a larger duration than that of the earthquakes. On the other hand, P picks are acceptable, but transfer learning fails in the event detection part (detections time span is much smaller than it should be).

**Comment by sfpoveda** [comment]
It seems like weights are not updating. Because I compared both the pretrained weights and the retrained weights, and I found they are the same. How is this possible? They shouldn't change once I do the training, but they do not. 

I did what you told me to the "target_state_dict". And after that I load the EQTransformer model with the modified weights and proceed to retrain the model, but neither validation loss nor validation accuracy change with epochs. Do you have any idea why my pretrained weights are not being updated?

**Comment by yetinam** [comment]
Hey @sfpoveda ,
sorry for the late reply, I'm currently on field work. The issue is indeed curious and reminds me of #98. Do you also use `original_compatible=True`? Otherwise, what you should check whether for the weights the `requires_grad` parameter is set to true. In principle, this seems to be a more pytorch related issue than a SeisBench issue, so maybe you'll find something looking on that side. I'd still be curious what's causing it, so that the issue can be avoided in the future.
